File: 2.Prgramming fundamentals/kurs fundamental 2017/09.DictionariesLambdaAndLINQExercises/02.PhonebookUpgrade/PhonebookUpgrade.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace _02.PhonebookUpgrade
{
    class PhonebookUpgrade
    {
        static void Main(string[] args)
        {
            List<string> input = Console.ReadLine().Split(' ').ToList();

            SortedDictionary<string, string> nameAndNumber = new SortedDictionary<string, string>();

            while (input[0] != "END")
            {

                if (input[0].Equals("A"))
                {
                    string name = input[1];
                    string number = input[2];

                    nameAndNumber[name] = number;
                }
                else if (input[0].Equals("S"))
                {
                    string name = input[1];

                    if (nameAndNumber.ContainsKey(name))
                        Console.WriteLine($"{name} -> {nameAndNumber[name]}");
                    else
                        Console.WriteLine($"Contact {name} does not exist.");
                }
                else if (input[0].Equals("ListAll"))
                {
                    foreach (var item in nameAndNumber)
                    {
                        Console.WriteLine(item.Key + " -> " + item.Value);
                    }
                }

                input = Console.ReadLine().Split(' ').ToList();
            }

        }
    }
}

File: 3.Software Technologies/00.SOFTWARE TECHNOLOGIES 2017/16.JavaSintaxBasicWebExercises/src/PhonebookUpgrade.java
import java.util.*;

public class PhonebookUpgrade {

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        TreeMap<String, String> phoneBook = new TreeMap<String, String>();

        String[] input = scanner.nextLine().split("\\s");

        while(!input[0].equals("END"))
        {
            String command = input[0];


            if(command.equals("ListAll"))
            {
                for (Map.Entry<String, String> kvp : phoneBook.entrySet()) {
                    System.out.println( kvp.getKey() + " -> " + kvp.getValue());
                }
            }

            if(command.equals("A"))
            {
                String name = input[1];
                String number = input[2];


                phoneBook.put(name,number);

            }
            else if(command.equals("S"))
            {

                String name = input[1];
                if (phoneBook.containsKey(name)) {
                    String phone = phoneBook.get(name);
                    System.out.println(name + " -> " + phone);
                }
                else
                {
                    System.out.println("Contact " + name + " does not exist.");
                }

            }



            input = scanner.nextLine().split("\\s");

        }


    }

}
